docs/developers/feature-versioning.md

```diff
@@ -0,0 +1,131 @@
+# Feature Versioning
+
+There are two types of features in Tekton Pipeline, API-driven features and behavioural features. Both types of features comply to the [feature gates](../../api_compatibility_policy.md#feature-gates) and the [feature graduation process](../../api_compatibility_policy.md#feature-graduation-process) specified in the [API compatibility policy](../../api_compatibility_policy.md).
+  - **API-driven features**: Features that are enabled via a specific field in pipeline API. Prior to the switch to per-feature flags, all features gated by `enable-api-fields` are considered as API-driven features. For example, [remote tasks](https://github.com/tektoncd/pipeline/blob/454bfd340d102f16f4f2838cf4487198537e3cfa/docs/taskruns.md#remote-tasks) is an API-driven feature.
+
+  - **Behavioural features**: Features that are not controlled by any specific API fields are considered as behavioural features. They are guarded behind their dedicated behavioural flags. For example, [`results-from`](https://github.com/tektoncd/pipeline/blob/fd4cb4621960715439bf0c0757f5be9cad390568/config/config-feature-flags.yaml#L112C3-L112C38) is a behavioural feature flag.
+
+## API-driven Features
```

Comment:
are we going to add a section of 'ehavioural features'?

Reply:
Yeah, as a reader I'm expecting to see a "behavioural feature" section in the doc, and I'm assuming all the content in this section is not applicable to "behavioural features" (but in fact, there are quite some similarity).